rumq-core/src/lib.rs

```diff
@@ -110,6 +110,7 @@ pub fn connect_return(num: u8) -> Result<ConnectReturnCode, Error> {
 
 #[derive(Debug, From)]
 pub enum Error {
+    #[from(ignore)]
```

Comment:
Why do we need this ignore?

Reply:
Oh this is to prevent `From u8` conflict..I wonder how the old version of `derive-more` worked

Reply:
Let's do this instead

```
#[derive(Debug, From)]
pub enum Error {
    InvalidConnectReturnCode(u8),
    InvalidProtocolName(String),
    InvalidProtocolLevel(String, u8),
    IncorrectPacketFormat,
    InvalidTopic,
    UnsupportedProtocolName,
    UnsupportedProtocolVersion,
    UnsupportedQoS,
    UnsupportedPacketType(u8),
    UnsupportedConnectReturnCode,
    PayloadSizeIncorrect,
    PayloadTooLong,
    PayloadRequired,
    #[from]
    TopicNameMustNotContainNonUtf8(FromUtf8Error),
    TopicNameMustNotContainWildcard,
    MalformedRemainingLength,
    UnexpectedEof,
    #[from]
    Io(io::Error),
}

```